### PR TITLE
Focus documentation

### PR DESCRIPTION
**Pull request by katarzynatobis**
Documentation of focus:
- styles (and description)
- mixins
- tokens

Screenshot 2022-11-07 at 13 37 46

### CODE REVIEW

File: src/docs/stories/focus.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    <Button>Button</Button>
+  </Flex>
+  <Flex fullWidth direction="column" alignItems="flex-start">
+    <Text weight="bold" style={{marginBottom: '8px'}}>
+      Focused
+    </Text>
+    <Button
+      style={{
+        boxShadow:
+          '0px 0px 0px 2px white, 0px 0px 0px 4px #6d83f3, 0px 0px 0px 6px rgba(109, 131, 243, 0.3)',
```
Comment (domidomi) [inline src/docs/stories/focus.stories.mdx:34]: @katarzynatobis you can use css variable here for indigo
```suggestion
          '0px 0px 0px 2px white, 0px 0px 0px 4px var(--indigo-40), 0px 0px 0px 6px rgba(109, 131, 243, 0.3)',
```

File: src/docs/stories/focus.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  <Flex fullWidth direction="column" alignItems="flex-start">
+    <Text weight="bold" style={{marginBottom: '8px'}}>
+      Focused
+    </Text>
+    <Link
+      style={{
+        outlineWidth: '2px',
+        outlineOffset: '4px',
+        outlineStyle: 'auto',
+        outlineColor: '#6d83f3',
```
Comment (domidomi) [inline src/docs/stories/focus.stories.mdx:71]: @katarzynatobis you can use css var here
```suggestion
        outlineColor: 'var(--indigo-40)',
```

File: src/docs/stories/focus.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+By default, the focus style is applied to almost all interactive elements (besides `<a>` and `<Link>`). For more complex cases (such as `:focus-within`), mixins can be used.
+
+| Mixin name      | Description                                                     |
+| --------------- | --------------------------------------------------------------- |
+| applyFocusStyle | The main focus style                                            |
+| applyFocus      | The main focus style and css logic when focus should be visible |
+
+## Focus text style
+
+In some cases, default focus style cannot be applied. For inline, multi-line elements a focus text style should used. By default, the focus text style is used by `<a>` and `<Link>`. It is based on the browser focus style.
```
Comment (coderitual) [inline src/docs/stories/focus.stories.mdx:53]: Can we rephrase this to be clear where we are meaning specific concepts and when generic ones?

E.g. Under the hood, the default focus style provided by the style guide (`applyFocusStyle`) uses a box shadow, which cannot always be used.

File: src/docs/stories/focus.stories.mdx (hunk 1 above)
Comment (katarzynatobis) [reply src/docs/stories/focus.stories.mdx:34]: 👍🏻

File: src/docs/stories/focus.stories.mdx (hunk 2 above)
Comment (katarzynatobis) [reply src/docs/stories/focus.stories.mdx:71]: 👍🏻

File: src/docs/stories/focus.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import Link from '../../components/text/Link';
+import Button from '../../components/buttons/Button';
+import Flex from '../../components/flex/Flex';
+import PageHeader from 'blocks/PageHeader';
+
+<Meta title="Foundation ✨/Focus" />
+
+<PageHeader type="foundation">Focus</PageHeader>
+
+Focus style is a visual indicators used to style elements, that have been focused by the user, either by "tabbing" using the keyboard or by clicking. Focus style helps people identify interactive elements.
```
[the remark this answers is not in the document]
Comment (katarzynatobis) [reply src/docs/stories/focus.stories.mdx:13]: That sounds a little bit better, thanks

File: src/docs/stories/focus.stories.mdx (hunk 3 above)
Comment (katarzynatobis) [reply src/docs/stories/focus.stories.mdx]: I tried to make it more understandable:

Under the hood, the default focus style provided by the style guide ([the focus style](#focus-style)) uses a box shadow, which may not always be used. For inline, multi-line elements you can use the focus text style, based on browser native focus style (with outline). By default, the focus text style is used by `<a>` and `<Link>`.

**Review by coderitual — APPROVED** [review]

File: src/docs/stories/focus.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    </Link>
+  </Flex>
+</Flex>
+
+### Mixins
+
+| Mixin name          | Description                                                                              |
+| ------------------- | ---------------------------------------------------------------------------------------- |
+| applyFocusTextStyle | Focus style for inline, multi-line components                                            |
+| applyFocusText      | Focus style for inline, multi-line components and css logic when focus should be visible |
```
Comment (clxandstuff) [inline src/docs/stories/focus.stories.mdx:85]: Can we use uppercase for CSS as for other acronyms?

**Review by clxandstuff — APPROVED** [review]
